packages/switch/src/mwc-switch.ts

```diff
@@ -0,0 +1,90 @@
+/**
+@license
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {FormElement, html, property, observer, query, customElement, BaseAdapter} from '@material/mwc-base/form-element';
+import {style} from './mwc-switch-css';
+import MDCSwitchFoundation from '@material/switch/foundation';
+
+export class SwitchAdapter extends BaseAdapter {
+  constructor(readonly element: Switch) {
```

Comment:
do you need the constructor? I think you can replace it with a type parameter on the base class for the type of `element`.

Reply:
I do need the element assigned somewhere though